lib/saml_idp_constants.rb

```diff
@@ -5,11 +5,6 @@ module Constants
       LOA1_AUTHNCONTEXT_CLASSREF = 'http://idmanagement.gov/ns/assurance/loa/1'.freeze
       LOA3_AUTHNCONTEXT_CLASSREF = 'http://idmanagement.gov/ns/assurance/loa/3'.freeze
       REQUESTED_ATTRIBUTES_CLASSREF = 'http://idmanagement.gov/ns/requested_attributes?ReqAttr='.freeze
-
```

Comment:
You're right! Not sure why I thought otherwise. I think a while back on another project I had used memoization with a constant and rubocop complained about that but didn't say that it was because memoization was not necessary. Maybe that is what caused my confusion.

Anyways, fix coming up.

Here's how I tested this:

I added a method to `Saml::Idp::Constants` with a `puts` statement, then made the constant call the method, and verified that the `puts` was only called once when running the tests.

```ruby
module Saml
  module Idp
    module Constants
      LOA1_AUTHN_CONTEXT_CLASSREF = 'http://idmanagement.gov/ns/assurance/loa/1'.freeze
      LOA3_AUTHN_CONTEXT_CLASSREF = 'http://idmanagement.gov/ns/assurance/loa/3'.freeze
      REQUESTED_ATTRIBUTES_CLASSREF = 'http://idmanagement.gov/ns/requested_attributes?ReqAttr='.freeze

      def self.authn_contexts
        puts 'called'
        JSON.parse(Figaro.env.valid_authn_contexts).freeze
      end

      VALID_AUTHN_CONTEXTS = Saml::Idp::Constants.authn_contexts
    end
  end
end
```